src/parser/warlock/demonology/modules/azerite/ShadowsBite.js

```diff
@@ -1,31 +1,100 @@
 import React from 'react';
 
-import Analyzer from 'parser/core/Analyzer';
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import calculateBonusAzeriteDamage from 'parser/core/calculateBonusAzeriteDamage';
+import StatTracker from 'parser/shared/modules/StatTracker';
 
 import SPELLS from 'common/SPELLS';
 import { formatThousands } from 'common/format';
+import { calculateAzeriteEffects } from 'common/stats';
 
 import TraitStatisticBox from 'interface/others/TraitStatisticBox';
 import ItemDamageDone from 'interface/others/ItemDamageDone';
 
-import ShadowsBiteForbiddenKnowledgeCore from './ShadowsBiteForbiddenKnowledgeCore';
+const DEMONBOLT_SP_COEFFICIENT = 0.667;
+const MAX_TRAVEL_TIME = 2000;
+const debug = false;
 
 class ShadowsBite extends Analyzer {
   static dependencies = {
-    core: ShadowsBiteForbiddenKnowledgeCore,
+    statTracker: StatTracker,
   };
+  bonus = 0;
+  damage = 0;
 
+  _queue = [
+    /*
+      {
+        timestamp: number
+        targetID: number,
+        targetInstance: number,
+        applySB: boolean
+        intellect: number
+      }
+     */
+  ];
   constructor(...args) {
     super(...args);
     this.active = this.selectedCombatant.hasTrait(SPELLS.SHADOWS_BITE.id);
+    if (!this.active) {
+      return;
+    }
+
+    this.bonus = this.selectedCombatant.traitsBySpellId[SPELLS.SHADOWS_BITE.id]
+      .reduce((total, rank) => {
+        const [ damage ] = calculateAzeriteEffects(SPELLS.SHADOWS_BITE.id, rank);
+        debug && this.log(`Rank ${rank}, damage ${damage}`);
+        return total + damage;
+      }, 0);
+
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.DEMONBOLT), this.onDemonboltCast);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER).spell(SPELLS.DEMONBOLT), this.onDemonboltDamage);
+  }
+
+  onDemonboltCast(event) {
+    // Shadow's Bite snapshots the bonus on cast
+    this._queue.push({
+      timestamp: event.timestamp,
+      targetID: event.targetID,
+      targetInstance: event.targetInstance,
+      applySB: this.selectedCombatant.hasBuff(SPELLS.SHADOWS_BITE_BUFF.id),
+      intellect: this.statTracker.currentIntellectRating,
```

Comment:
Rather than relying on `statTracker` to be correct you can access the player's spell power directly from `event.spellPower` for a cast event such as this.

Reply:
This might reveal that we have a more severe problem in StatTracker as a whole, that we might be double-dipping the on-pull buffs all the time. The log in the link shows the player's intellect as 9198, which is indeed passed as `_pullStats`:
`StatTracker: 00:00.000 - STR=557 AGI=1130 INT=9198 STM=9562 CRT=1868 HST=1055 MST=795 VRS=184 AVD=0 LCH=0 SPD=0 ARMOR=1142`

But then, the buff normalizer fabricates applybuffs for Intel flask, Intel food, which gets picked up again
```
StatTracker: (0 -> 1) Flask of Endless Fathoms @ 00:00.000 - Change: STR=0 AGI=0 INT=238 STM=0 CRT=0 HST=0 MST=0 VRS=0 AVD=0 LCH=0 SPD=0 ARMOR=1142 
StatTracker: 00:00.000 - STR=557 AGI=1130 INT=9436 STM=9562 CRT=1868 HST=1055 MST=795 VRS=184 AVD=0 LCH=0 SPD=0 ARMOR=1142 
StatTracker: (0 -> 1) Bountiful Captain's Feast @ 00:00.000 - Change: STR=0 AGI=0 INT=100 STM=0 CRT=0 HST=0 MST=0 VRS=0 AVD=0 LCH=0 SPD=0 ARMOR=1142 
StatTracker: 00:00.000 - STR=557 AGI=1130 INT=9536 STM=9562 CRT=1868 HST=1055 MST=795 VRS=184 AVD=0 LCH=0 SPD=0 ARMOR=1142
```
resulting in 338 more intellect than it should have (*maybe*).

At this point, I'm conflicted as what to believe - do the `_pullStats` contain all the buffs already? Is `event.spellPower` accurate?